File: backend/src/PxgTools.Domain/Pokemon/Pokemon.cs
namespace PxgTools.Domain.Pokemon;

public sealed record PokemonListItem(
    string Slug,
    int DexNumber,
    string Dex,
    string Name,
    string Generation,
    string SpriteUrl,
    string SourceUrl,
    string Level,
    string Boost,
    string Material,
    IReadOnlyList<string> Elements);

public sealed record PokemonEvolution(
    string Name,
    string Level);

public sealed record PokemonEffectivenessGroup(
    string Category,
    IReadOnlyList<string> Types);

public sealed record PokemonMove(
    string Name,
    string Type,
    string Cooldown,
    string Level,
    string Description);

public sealed record PokemonVersion(
    string Name,
    string Slug,
    string IconUrl,
    string SourceUrl);

public sealed record PokemonDetail(
    string Slug,
    int DexNumber,
    string Dex,
    string Name,
    string Generation,
    string SpriteUrl,
    string DetailSpriteUrl,
    string SourceUrl,
    string Level,
    IReadOnlyList<string> Elements,
    string Abilities,
    string Boost,
    string Material,
    string EvolutionStone,
    IReadOnlyList<PokemonEvolution> Evolutions,
    string Description,
    IReadOnlyList<PokemonEffectivenessGroup> Effectiveness,
    IReadOnlyList<PokemonMove> PvpMoves,
    IReadOnlyList<PokemonMove> PveMoves,
    IReadOnlyList<PokemonVersion> OtherVersions);

public sealed record PokemonOverview(
    IReadOnlyList<string> Generations,
    IReadOnlyList<PokemonListItem> Pokemon);

File: frontend/src/types/pokemon.ts
export type PokemonListItem = {
  slug: string;
  dexNumber: number;
  dex: string;
  name: string;
  generation: string;
  spriteUrl: string;
  sourceUrl: string;
  level: string;
  boost: string;
  material: string;
  elements: string[];
  clanNames?: string[];
};

export type PokemonEvolution = {
  name: string;
  level: string;
};

export type PokemonEffectivenessGroup = {
  category: string;
  types: string[];
};

export type PokemonMove = {
  name: string;
  type: string;
  cooldown: string;
  level: string;
  description: string;
  icons: PokemonMoveIcon[];
};

export type PokemonMoveIcon = {
  label: string;
  iconUrl: string;
};

export type PokemonVersion = {
  name: string;
  slug: string;
  iconUrl: string;
  sourceUrl: string;
};

export type PokemonLootItem = {
  itemName: string;
  itemNameEn: string;
  itemNamePtBr: string;
  itemSlug: string;
  iconUrl: string;
  category: string;
  sourceUrl: string;
  pokemonName: string;
  pokemonSlug: string;
  isVariant: boolean;
};

export type PokemonDetail = PokemonListItem & {
  detailSpriteUrl: string;
  abilities: string;
  boost: string;
  material: string;
  evolutionStone: string;
  evolutions: PokemonEvolution[];
  description: string;
  effectiveness: PokemonEffectivenessGroup[];
  moves?: PokemonMove[];
  pvpMoves: PokemonMove[];
  pveMoves: PokemonMove[];
  otherVersions: PokemonVersion[];
  loot?: PokemonLootItem[];
};

export type PokemonOverview = {
  generations: string[];
  pokemon: PokemonListItem[];
};
